wagtail_modeltranslation/static/wagtail_modeltranslation/js/language_toggles.js

```diff
@@ -0,0 +1,117 @@
+///////////////
+jQuery( () => {
+///////////////
+
+const tabbedContent = $(`form .tab-content`);
+const topLevel = (tabbedContent.length > 0) ? tabbedContent : $(`.content form`);
+
+if (topLevel.length === 0) {
+  // obviously, if we don't have an element to attach
+  // the picker to, we might as well stop right now.
+  return;
+}
+
+if (topLevel.attr(`class`) && topLevel.attr(`class`).indexOf(`search`) > -1) {
+  // if the only forms on the page are search forms,
+  // we're not actually dealing with page/snippets
+  return;
+}
+
+/**
+ * Build the set of fields-per-locale. Each set will receive
+ * a button to toggle visibility for all fields in that set,
+ * with the note that unlocalised content (such as images)
+ * will always stay visible.
+ */
+function buildSets() {
+  // Note: we do NOT rely on the "langs" variable
+  // that wagtail-modeltranslations sets up, as
+  // we want to know whether "this page",
+  // specifically, has any localised fields that
+  // require set building, rather than which URL
+  // language prefixes are supported, in general.
+  var localisedElements = {};
+
+  $(`li.object`, topLevel).each( (index, element) => {
+    var tc = element.textContent;
+    var res = tc.match(/\[\w\w\]/);
+    if (res === null) return;
+
+    var locale = res[0].replace(/[\[\]]/g,'');
+
+    if (!localisedElements[locale]) {
+      localisedElements[locale] = [];
+    }
+
+    localisedElements[locale].push(element);
+    element.classList.add(`l10n-hidden`);
+  });
+
+  return localisedElements;
+}
+
+/**
+ * Build a locale picker bar, with buttons that toggle
+ * visibility for each locale's fields.
+ */
+function buildLocaleToggler() {
+  var bar = $(`<div class="locale-picker"><h2>View/edit fields for:</h2></div>`);
+  var ul = $(`<ul class="locales"></ul>`);
+  bar.append(ul);
+
+  var toggles = {};
+  locales.forEach( locale => {
+    var li = $(`<li class="locale"><button class="locale-toggle">${locale}</button></li>`);
+    ul.append(li);
+
+    $(`button.locale-toggle`, li).each( (index, toggle) => {
+      toggle.addEventListener(`click`, e => {
+        e.preventDefault();
+        toggle.classList.toggle(`active`);
+        toggleLocale(locale);
+      })
+
+      toggles[locale] = toggle;
+    });
+  });
+
+  bar.prependTo(topLevel);
+
+  return toggles;
+}
+
+/**
+ * This function allows either blind toggling
+ * of a field's visibility, or explicitly
+ * making visible/invisible based on the
+ * value of `state` (a boolean).
+ */
+function toggleLocale(locale, state) {
+  var action = `toggle`;
+
+  if (state !== undefined) {
+    action = state ? `remove` : `add`
+  }
+
+  localisedElements[locale].forEach(element => {
+    element.classList[action](`l10n-hidden`);
+  });
+}
+
+var default_locale = `en`;
```

Comment:
this is the only part I'm not sure about: there is no global scope variable to consult to see what the `DEFAULT_LANGUAGE` variable is set to, and defaulting to English, which might not even be in the `LANGUAGES` set, seems more than a little presumptuous.

Reply:
Hello @Pomax 
Great work.
I agree that this is a problem. I need some time to think about a way to pass this variable to JS.
@Alexandre27 @dmarcelino any ideas?

Cheers

Reply:
it might make sense to add it in the same place that `langs` is injected right now. Maybe make it something like this?

```
wagtailModelTranslations = {
   languages: <from LANGUAGES>,
  defaultLanguage: <from LANGUAGE_CODE>,
}
```

At the very least that would also mean that there's no global namespace clobbering in case someone's script uses a var `langs` (I didn't know there was a `langs` until I dug into the hooks file  and noticed it got injected =D)

The other thing this'll need is some way to localize the actual strings used by this script. There's not a lot of them, but "View / edit fields for" definitely shouldn't be hard-coded as English string.

Reply:
Just added some commits to implement most of this.

About the strings translation, I think we should do something similar to what we've done here so we can register them in the backend for translation.